doc/rules/control_structure/fully_multiline.rst

```diff
@@ -0,0 +1,94 @@
+========================
+Rule ``fully_multiline``
```

Comment:
Not sure about the name of the rule.

It will follows the PSR12 rule for control structure:
> expressions in parentheses MAY be split across multiple lines. When doing so, the first condition MUST be on the next line.

But also solve https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6675 for arrays, switch cases, match expression by allowing only one expression per line.

Reply:
There are many rules with `single_*_per_*` naming convention, so maybe `single_expression_per_line`?

I'll do my best to look more at your PR soon, but I have `#7777` to finish, also created `#7815` while waiting for some feedback, so a lot to do with higher priority (especially the former), and my time is limited 😅. Thanks for your understanding.